### PR TITLE
Update text-fonts.md page to v3 methods

### PR DESCRIPTION
**Pull request by miguilimzero**
This PR fixes the font methods in the documentation. It seems it's currently showing the v2 one.

- Updates methods to `set...`.
- Updates `align` and `valign` to `setAlignment` and `setValignment`.

### CODE REVIEW

File: image/v3/modifying/text-fonts.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ a callback as the fourth parameter.
 | text | string | The text that will be written to the image. |
 | x | integer | Coordinate on x-axis defining the base point of the first character. |
 | y | integer | Coordinate on y-axis defining the base point of the first character. |
-| settings | callable | Callback function to configure the font appearance. |
+| font | callable|FontInterface | Callback function to configure the font appearance or `Typography\Font` instance. |
```
Comment (olivervogel) [inline image/v3/modifying/text-fonts.md:21]:
```suggestion
| font | callable or FontInterface | Callback function to configure the font appearance or `Typography\Font` instance. |
```
The pipe character interferes unfortunately with table generation.

File: image/v3/modifying/text-fonts.md [hunk 2]
```diff
@@ -49,21 +49,21 @@ $image = ImageManager::imagick()->read('images/example.jpg');
...
-    $font->valign('middle');
-    $font->lineHeight(1.6);
-    $font->angle(10);
+    $font->setFilename('./fonts/comic-sans.ttf');
+    $font->setColor('#b01735');
+    $font->setSize(70);
+    $font->setAlignment('center');
+    $font->setValignment('middle');
+    $font->setLineHeight(1.6);
+    $font->setAngle(10);
```
Comment (olivervogel) [inline image/v3/modifying/text-fonts.md:58]: Please revert. These were correct calls of class `Intervention\Image\Typography\FontFactory::class`.

File: image/v3/modifying/text-fonts.md [hunk 3]
```diff
@@ -37,7 +37,7 @@ $image->text('The quick brown fox', 120, 100);
 
 To define the overall appearance of the text and set more details you can pass
 a callback as an optional parameter. The callback places the calls on the
-FontInterface and listens to the following methods.
+FontInterface and listen to the following methods.
```
Comment (olivervogel) [inline image/v3/modifying/text-fonts.md:40]:
```suggestion
FontInterface and listens for following methods.
```
I think "listens" is correct. But "listens for" is better.

File: image/v3/modifying/text-fonts.md [hunk 4]
```diff
@@ -49,21 +49,21 @@ $image = ImageManager::imagick()->read('images/example.jpg');
...
+    $font->setValignment('middle');
+    $font->setLineHeight(1.6);
+    $font->setAngle(10);
 });
 ```
 
 ### Font size
 
-> public FontInterface::size(float $size): FontInterface
+> public FontInterface::setSize(float $size): FontInterface
```
Comment (olivervogel) [inline image/v3/modifying/text-fonts.md:64]: Please revert. This is correct call of class Intervention\Image\Typography\FontFactory::class.

File: image/v3/modifying/text-fonts.md [hunk 5]
```diff
@@ -73,9 +73,9 @@ Define a font size. By default a value of `12` will be applied.
 
 ### Font file
 
-> public FontInterface::filename(string $filename): FontInterface
+> public FontInterface::setFilename(string $filename): FontInterface
```
Comment (olivervogel) [inline image/v3/modifying/text-fonts.md:76]: Please revert. This is correct call of class Intervention\Image\Typography\FontFactory::class.

File: image/v3/modifying/text-fonts.md [hunk 6]
```diff
@@ -85,7 +85,7 @@ Set a path to a font file in file system in which the text should be written.
 
 ### Color
 
-> public FontInterface::color(mixed $color): FontInterface
+> public FontInterface::setColor(mixed $color): FontInterface
```
Comment (olivervogel) [inline image/v3/modifying/text-fonts.md:88]: Please revert. This is correct call of class Intervention\Image\Typography\FontFactory::class.

File: image/v3/modifying/text-fonts.md [hunk 7]
```diff
@@ -97,7 +97,7 @@ Define the text color in one of the valid [color formats](/v3/introduction/forma
 
 ### Horizontal alignment
 
-> public FontInterface::align(string $align): FontInterface
+> public FontInterface::setAlignment(string $align): FontInterface
```
Comment (olivervogel) [inline image/v3/modifying/text-fonts.md:100]: Please revert. This is correct call of class Intervention\Image\Typography\FontFactory::class.

File: image/v3/modifying/text-fonts.md [hunk 8]
```diff
@@ -110,7 +110,7 @@ base point. Possible values are left, right and center. Default: `left`
 
 ### Vertical alignment
 
-> public FontInterface::valign(string $valign): FontInterface
+> public FontInterface::setValignment(string $valign): FontInterface
```
Comment (olivervogel) [inline image/v3/modifying/text-fonts.md:113]: Please revert. This is correct call of class Intervention\Image\Typography\FontFactory::class.

File: image/v3/modifying/text-fonts.md [hunk 9]
```diff
@@ -123,7 +123,7 @@ point. Possible values are top, bottom and middle. Default: `bottom`
 
 ### Rotation
 
-> public FontInterface::angle(float $angle): FontInterface
+> public FontInterface::setAngle(float $angle): FontInterface
```
Comment (olivervogel) [inline image/v3/modifying/text-fonts.md:126]: Please revert. This is correct call of class Intervention\Image\Typography\FontFactory::class.

File: image/v3/modifying/text-fonts.md [hunk 10]
```diff
@@ -135,13 +135,13 @@ Rotate the text block clockwise with a desired angle.
 
 ### Line height
 
-> public FontInterface::lineHeight(float $height): FontInterface
+> public FontInterface::setLineHeight(float $height): FontInterface
```
Comment (olivervogel) [inline image/v3/modifying/text-fonts.md:138]: Please revert. This is correct call of class Intervention\Image\Typography\FontFactory::class.

**Review by olivervogel — CHANGES_REQUESTED** [review]

**Comment by olivervogel** [comment]
Thank you. Some of the updates are not necessary, as they are calls from special helper classes.

**Comment by miguilimzero** [comment]
Sorry @olivervogel. Totally my confusion. When I was upgrading my project to v3, I accidentally used `Font` to type the callable parameter, and the psalm accused the methods were wrong, making me investigate the classes and think that the documentation might be outdated. Sorry for you spending your time on this.

But another thing that made me think the documentation might be wrong is that the documentation says these methods are from the `FontInterfance` and not `FontFactory`. It's currently showing:

```md
### Font size

> public FontInterface::size(float $size): FontInterface
```

Shouldn't it be something like that:

```md
### Font size

> public FontFactory::size(float $size): FontFactory
```

**Comment by olivervogel** [comment]
> But another thing that made me think the documentation might be wrong is that the documentation says these methods are from the `FontInterfance` and not `FontFactory`.

Yes, you are right.

**Pull request merged**